fastlane/lanes/release.rb

```diff
@@ -157,18 +304,39 @@
   desc 'Trigger the final release build on CI'
   lane :finalize_release do |options|
     UI.user_error!('To finalize a hotfix, please use the finalize_hotfix_release lane instead') if ios_current_branch_is_hotfix
+    ensure_git_branch(branch: '^release/') # Match branch names that begin with `release/`
+    ensure_git_status_clean
+
+    message = "Finalizing release: #{current_release_version}\n"
+    if options[:skip_confirm]
+      UI.message(message)
+    else
+      UI.user_error!('Aborted by user request') unless UI.confirm("#{message}Do you want to continue?")
+    end
 
-    ios_finalize_prechecks(options)
     git_pull
 
     check_all_translations(interactive: true)
 
     download_localized_strings_and_metadata(options)
     ios_lint_localizations(input_dir: 'WordPress/Resources', allow_retry: true)
-    ios_bump_version_beta
+
+    # Bump the build code
+    UI.message 'Bumping build code...'
+    ensure_git_branch(branch: '^release/') # Match branch names that begin with `release/`
+    PUBLIC_VERSION_FILE.write(version_long: next_build_code)
+    Fastlane::Helper::Ios::GitHelper.commit_version_bump
+    UI.message "Done! New Build Code: #{current_build_code}"
+
+    # Bump the internal build code
+    UI.message 'Bumping internal build code...'
+    INTERNAL_VERSION_FILE.write(version_long: next_internal_build_code)
+    UI.message "Done! New Internal Build Code: #{current_internal_build_code}"
+
+    Fastlane::Helper::Ios::GitHelper.commit_version_bump
```

Comment:
There is a minor inconsistency here between when the message is printed vs the commit made.

Personally, I prefer to commit first and print the confirmation as the very last step:

```suggestion
    Fastlane::Helper::Ios::GitHelper.commit_version_bump
    UI.message "Done! New Build Code: #{current_build_code}"

    # Bump the internal build code
    UI.message 'Bumping internal build code...'
    INTERNAL_VERSION_FILE.write(version_long: next_internal_build_code)
    Fastlane::Helper::Ios::GitHelper.commit_version_bump
    UI.message "Done! New Internal Build Code: #{current_internal_build_code}"
```

Reply:
Related: is it necessary to commit twice? Currently, we only do it once and commit both files, e.g. acf08a0dae55197dbdf2fd12d4f09aecc3f29f13

Reply:
Addressed in https://github.com/wordpress-mobile/WordPress-iOS/pull/21666/commits/75334b7c2ba858e1da781ee0d0f95c0ab7805d06